js/interface.js

```diff
@@ -1208,8 +1208,35 @@ var DynamicLists = (function() {
         createTokensOnBlur: true
       });
 
+      _this.handleTokensSelection();
       _this.loadTokenFields();
     },
+    handleTokensSelection: function () {
+      $('input.tokenfield').on('tokenfield:createdtoken tokenfield:removedtoken', function (event) {
+        var field = $(this);
+        var currentTokens = field.tokenfield('getTokens');
+        var originalSource = field.data('bs.tokenfield').options.autocomplete.source;
+
+        // Clone original autocomplete source
+        var newSource = originalSource.slice();
+        for (var i = newSource.length - 1; i >= 0; i--) {
+          for (var j = currentTokens.length - 1; j >= 0; j--) {
+            if (_.isEqual(currentTokens[j].label, newSource[i]) || _.isEqual(currentTokens[j], newSource[i])) {
+
+              // Remove the token from the newSource
+              var index = newSource.indexOf(newSource[i]);
+              if (index > -1) {
+                newSource.splice(index, 1);
```

Comment:
Can we not use `_.remove` and then use the predicate function to run the comparison?

We can also use `_.some` so that the loop doesn't run for longer than necessary, right?